### PR TITLE
Feature/templates config

### PR DESCRIPTION
**Pull request by Kaydayo**
## Description

[Describe the change that was made in this PR]

### Todos

- [x ] Tested and working locally
- [ ] Unit tests added (if appropriate)
- [ ] Code changes documented
- [x] Requested review from >=1 devs on the team

### How to test

[Add detailed steps to test the changes that have been made]

### Screenshots and/or Gifs

[Insert screenshots and/or gifs showing the graphic representation of the change]

### Associated JIRA Tasks

- [Task Code: Task Description](http://jira_task_link.com)

### Known Issues

[Describe any known issue with this change]

### CODE REVIEW

File: src/company/integrations/integrations.controller.ts [hunk 1]
```diff
@@ -19,9 +20,10 @@ export class IntegrationsController {
   @Get(':type/init')
   async initIntegration(
     @Param() params: InitIntegrationDto,
```
Comment (ayopedro) [inline src/company/integrations/integrations.controller.ts:22]: The param here is referring to the `:type` right? if yes, you can just pass it into the param decorator 

```
@Param('type') type: string

```

File: src/company/integrations/integrations.controller.ts [hunk 2]
```diff
@@ -19,9 +20,10 @@ export class IntegrationsController {
...
+    @Query() query: { code: string },
     @Req() req: RequestWithUser,
```
Comment (ayopedro) [inline src/company/integrations/integrations.controller.ts:24]: you can use a DTO for the query

File: src/company/integrations/integrations.controller.ts [hunk 3]
```diff
@@ -19,9 +20,10 @@ export class IntegrationsController {
...
   ) {
-    const data = { ...req.user, type: params?.type };
+    const data = { ...req.user, type: params?.type, companyCode: query?.code };
     return this.integrationsService.handleIntegrationRequest(
```
Comment (ayopedro) [inline src/company/integrations/integrations.controller.ts:27]: this function can just receive the type and the dto (data)

File: src/company/integrations/integrations.service.ts [hunk 4]
```diff
@@ -67,23 +74,64 @@ export class IntegrationsService extends CrudService<
         });
         result = true;
       }
+
+      const maxAge = 24 * 60 * 60 * 1000;
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:78]: magic value here

Comment (ayopedro) [reply src/company/integrations/integrations.service.ts:78]: I added a method in the AppUtilities which converts seconds to milliseconds so you can use that

File: src/company/integrations/integrations.service.ts [hunk 5]
```diff
@@ -67,23 +74,64 @@ export class IntegrationsService extends CrudService<
...
-      throw new HttpException(
+      throw new UnprocessableEntityException(
         error.message || 'an error occurred',
-        HttpStatus.UNPROCESSABLE_ENTITY,
       );
     } finally {
-      return `${process.env.CLIENT_URL}/dashboard/account?type=${integration_type}&success=${result}`;
+      const getBaseClient = this.config.get<string>('app.clientUrl');
+      const protocol =
+        this.config.get<string>('environment') === 'prod'
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:107]:
```suggestion
        this.config.get<string>('environment') !== 'development'
```

File: src/company/integrations/integrations.service.ts [hunk 6]
```diff
@@ -67,23 +74,64 @@ export class IntegrationsService extends CrudService<
...
+    const provider = this.getIntegrationProvider(integration_type);
     const uri = provider.getIntegrationUri(payload);
     return { uri };
   }
+
+  public async getAllIntegrations(queryParam: IntegrationQuery) {
+    const query = buildIntegrationQuery(queryParam);
+    return await this.companyPrismaClient.integrationsConfig.findMany({
+      where: query,
+      distinct: ['integration_type'],
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:128]: use an enum for the integration_type and also, i think we should stick to camelCasing

**Review by ayopedro — CHANGES_REQUESTED** [review]

File: src/company/integrations/integrations.service.ts (hunk 6 above)
Comment (Kaydayo) [reply src/company/integrations/integrations.service.ts:128]: its an enum

File: src/company/integrations/integrations.controller.ts (hunk 1 above)
Comment (Kaydayo) [reply src/company/integrations/integrations.controller.ts:22]: need the type checking here hence the dto

File: src/company/integrations/integrations.controller.ts (hunk 3 above)
Comment (Kaydayo) [reply src/company/integrations/integrations.controller.ts:27]: yup, right

**Pull request merged**